src/routes/products.js

```diff
@@ -1,69 +1,129 @@
-          id: getProductById.id,
-          sku: getProductById.sku,
-          name: getProductById.name,
-          description: getProductById.description
-        };
-        productsArr.push(productObj);
-      }
-      if (productsArr.length === 0) {
-        foundedProducts = {
-          status: "no products",
-          products: []
-        };
-      } else {
-        foundedProducts = {
-          status: "success",
-          products: productsArr
-        };
-      }
+router.get("/:productId", (req, res) => {
+  const id = req.params.productId;
+  Product.findById(id)
+    .exec()
+    .then(item => {
+      console.log(item);
+      res.status(200).json({
+        status: "success",
+        product: item
+      });
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({ error: err });
     });
+});
 
-    res.setHeader("Content-Type", "application/json");
-    res.send(foundedProducts);
-    res.end();
-    return;
-  }
+router.get("/", (req, res) => {
+  Product.find()
+    .exec()
+    .then(items => {
+      console.log(items);
+      res.status(200).json(items);
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({ error: err });
+    });
+});
 
-  let oneProductId = req.url.slice(req.url.lastIndexOf("/") + 1);
-  const getOneProductById = products.find(
-    item => item.id.toString() === oneProductId
-  );
-  if (getOneProductById) {
-    res.setHeader("Content-Type", "application/json");
-    res.send(getOneProductById);
-    res.end();
-  } else {
-    res.setHeader("Content-Type", "application/json");
-    res.send(
-      JSON.stringify({
-        status: "no products",
-        products: []
+router.post("/", jsonParser, (req, res) => {
+  if (checkProduct(req.body)) {
+    console.log("Validation complete!");
+    const product = new Product({
+      _id: new mongoose.Types.ObjectId(),
+      sku: req.body.sku,
+      name: req.body.name,
+      description: req.body.description,
+      price: req.body.price,
+      currency: req.body.currency,
+      creatorId: req.body.creatorId,
+      created: req.body.created,
+      modified: req.body.modified,
+      categories: [
+        {
+          category: req.body.categories[0]
+        }
+      ],
+      likes: req.body.likes
+    });
+
+    product
+      .save()
+      .then(result => {
+        console.log(result);
+        res.status(201).json({
+          status: "success",
+          product: product
+        });
       })
-    );
-    res.end();
+      .catch(err => {
+        console.log(err);
+        res.status(500).json({ error: err });
+      });
+  } else {
+    console.log("Validation error!");
+    res
+      .status(500)
+      .json({ error: "failed, you must enter correct type of data" });
   }
 });
 
+router.put("/:productId", jsonParser, (req, res) => {
+  const id = req.params.productId;
+  const reqBody = req.body;
+  const newValueKey = Object.keys(reqBody).toString();
```

Comment:
Also it's not a good idea to do something like that. What would happen if user send multiple fields?